### DISCUSSION TITLE
Hide all Comments, Hide all print statements

### DISCUSSION THREAD

**Discussion by shak2** [godotengine/godot-proposals#9754] [Editor]
Fluff is distracting when you are trying to fix a deep bug, you want to see tight code only.

Imagine a shortcut to instantly hide all comment lines and all print lines (print, printt, prints etc).

It would be great to hide all print statement lines and hide all comment lines using 2 UI buttons in the top right of the gdscript script writing editor window. 

Or would this be better in the script window top left menu options? Edit - hide print statemnts, Edit - hide comments?

Shortcuts for these two things would also be nice.

Any ideas for what would be a good available default shortcut for these? (CTRL+c is taken)

**Comment by AThousandShips** [comment]
Hiding all comments might be useful, but print statements aren't decoration, they're functional, and might contain things that caused errors, so hiding them is very different IMO

**Reply by shak2** [reply]
RE: print statement hiding:

Print statements are technically functional, yes, but it is incredibly rare for them to be used to do anything other than to simply output variables (although they can be used for other purposes I have never seen them used for anything else).

For this reason, and to leave it to user choice, and that they are being hidden not disabled, I think hiding print statements should be added as a user ability.

**Reply by AThousandShips** [reply]
I've never needed to hide them myself, and if they contain something like `print(foo.bar)` it can easily have a bug, so I don't think many people will or even should use it

**Reply by shak2** [reply]
The print statements will be hidden not disabled! so if they do have a function that will still happen when you run the game. This hiding prints proposal would just remove their line bloat temporarily in the editor via a reversible shortcut.

**Reply by AThousandShips** [reply]
I didn't say to disable them, but it is very different from hiding comments, errors happen in print statements as well, and I don't see how debugging is changed by removing print statements, unless you use way too many of them

**Reply by shak2** [reply]
I use hundreds of print statements. I find print statements give instant feedback on code progress and can catch bugs before they happen, often making breakpoint style debugging unnecessary. It is for this reason that the option to hide them would be useful vs the alternative of individually commenting out a hundred print statements.

**Reply by AThousandShips** [reply]
That doesn't apply to most users so this feature wouldn't be useful for them, I'd suggest using breakpoints or other means, you're making your code slower, but your idea doesn't apply to most people

**Reply by theraot** [reply]
@shak2 If print statements are as prevalent in your code as you make them seem, then I would worry that while editing the code with print statements hidden the print statement would be rendered pointless, useless, or even turn into errors when they reference variables that changed name or no longer exists. I suppose the error is the lesser problem - conceptually - as the error should still be shown.

With the fetaure you propose I would worry, both for comments and for print statements, that they might become silently obsolete.

In fact, if I need to refactor such code, I would rather remove whatever prints and comments become obsolete so I can write them again afterwards (perhaps writing a TODO so I don't forget).

Thus, at least, it is not for me, yet... Why do you not share this worry of prints and comments becoming silently obsolete?

**Comment by shak2** [comment]
> That doesn't apply to most users so this feature wouldn't be useful for them

I disagree with this. 

I think the majority of users spam print messages to track code progress as they type, nearly every tutorial I have watched on YT has the teacher doing this at some point to verify output. It is in my opinion very common.

**Comment by Calinou** [comment]
I don't recall seeing any code editor out there that lets you hide all comments in a non-destructive manner, let alone print statements. In general, we try to align the Godot script editor's functionality on popular script editors, so that we avoid adding features that may not end up seeing much use.

**Reply by theraot** [reply]
I have seen extensions that change the color of comments to make them less visible as a toggle. I have also seen extensions that collapse/fold comments also as a toggle.

I think a command - that could be bind to a keyboartd shortcut of the user choice - to collapse/fold all comments would fit well in Godot.

---

But the closest I have seen to hiding prints is stripping them as a step before minimizing JavaScript (in fact, I once implemented a solution that removed lines depending on the comments, as a poor man's `#if` for JavaScript).

Another thing that is similar would be debug only calls in .NET (which are calls to any method with the appropiate annotation) that are excluded in release builds.

While something like that has merit, it is not what OP is asking.

**Reply by AThousandShips** [reply]
One thing we could do to make scanning through code quickly would be to highlight some methods like print a different colour from other methods, but unsure if it's worth the code complexity
